tests/interfaces/default_qubit_2_integration/test_torch_qnode_default_qubit_2.py

```diff
@@ -0,0 +1,2164 @@
+            pytest.skip("The hadamard method does not yet support Hamiltonians")
+
+        obs = [qml.PauliX(0), qml.PauliX(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliZ(1)]
+
+        @qnode(
+            dev,
+            diff_method=diff_method,
+            grad_on_execution=grad_on_execution,
+            max_diff=max_diff,
+            interface="torch",
+        )
+        def circuit(data, weights, coeffs):
+            weights = torch.reshape(weights, [1, -1])
+            qml.templates.AngleEmbedding(data, wires=[0, 1])
+            qml.templates.BasicEntanglerLayers(weights, wires=[0, 1])
+            return qml.expval(qml.Hamiltonian(coeffs, obs))
+
+        d = torch.tensor([0.1, 0.2], requires_grad=False, dtype=torch.float64)
+        w = torch.tensor([0.654, -0.734], requires_grad=True, dtype=torch.float64)
+        c = torch.tensor([-0.6543, 0.24, 0.54], requires_grad=True, dtype=torch.float64)
+
+        # test output
+        res = circuit(d, w, c)
+
+        expected = c[2] * torch.cos(d[1] + w[1]) - c[1] * torch.sin(d[0] + w[0]) * torch.sin(
+            d[1] + w[1]
+        )
+        assert torch.allclose(res, expected, atol=tol)
+
+        # test gradients
+        res.backward()
+        grad = (w.grad, c.grad)
+
+        expected_w = torch.tensor(
+            [
+                -c[1] * torch.cos(d[0] + w[0]) * torch.sin(d[1] + w[1]),
+                -c[1] * torch.cos(d[1] + w[1]) * torch.sin(d[0] + w[0])
+                - c[2] * torch.sin(d[1] + w[1]),
+            ]
+        )
+        expected_c = torch.tensor(
+            [0, -torch.sin(d[0] + w[0]) * torch.sin(d[1] + w[1]), torch.cos(d[1] + w[1])]
+        )
+        assert torch.allclose(grad[0], expected_w, atol=tol)
+        assert torch.allclose(grad[1], expected_c, atol=tol)
+
+        # test second-order derivatives
+        if diff_method in ("parameter-shift", "backprop") and max_diff == 2:
+            hessians = torch.autograd.functional.hessian(circuit, (d, w, c))
+
+            grad2_c = hessians[2][2]
+            assert torch.allclose(grad2_c, torch.zeros([3, 3], dtype=torch.float64), atol=tol)
+
+            grad2_w_c = hessians[1][2]
+            expected = torch.tensor(
+                [
+                    [0, -torch.cos(d[0] + w[0]) * torch.sin(d[1] + w[1]), 0],
+                    [
+                        0,
+                        -torch.cos(d[1] + w[1]) * torch.sin(d[0] + w[0]),
+                        -torch.sin(d[1] + w[1]),
+                    ],
+                ]
+            )
+            assert torch.allclose(grad2_w_c, expected, atol=tol)
+
+    @pytest.mark.parametrize("max_diff", [1, 2])
+    def test_hamiltonian_expansion_finite_shots(
+        self, dev, diff_method, grad_on_execution, max_diff
+    ):
+        """Test that the Hamiltonian is expanded if there
+        are non-commuting groups and the number of shots is finite
+        and the first and second order gradients are correctly evaluated"""
```

Comment:
This test doesn't actually check this. The new device doesn't use `hamiltonian_expand`, but it does calculate expectation values for each hamiltonian term separately. This is a valid test, but it might be a duplicate or belongs inside a different class inside this file.

Reply:
caught this on some interface test files, not all. updated in latest commit